core/src/main/java/com/exadel/etoolbox/backpack/core/dto/response/ResourceRelationships.java

```diff
@@ -0,0 +1,64 @@
+package com.exadel.etoolbox.backpack.core.dto.response;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Represents the wrapper for a collection of active paths and broken paths obtained as a result of the search.
+ */
+public class ResourceRelationships {
+
+    private List<String> actualPaths;
+    private List<String> brokenPaths;
+
+    public ResourceRelationships(List<String> actualPaths) {
+        this.actualPaths = actualPaths;
+    }
+
+    public ResourceRelationships(List<String> actualPaths, List<String> brokenPaths) {
+        this.actualPaths = actualPaths;
+        this.brokenPaths = brokenPaths;
+    }
+
+    public List<String> getActualPaths() {
+        return actualPaths != null ? actualPaths : new ArrayList<>();
```

Comment:
If I called this method 100 times, I would get 100 different instances of new ArrayList) Consider creating one at the time of class initialization, and then reusing it in the setter (like: `this.actualPaths.addAll(actualPaths))`). Or else maybe you don't need this setter at all, because there are already two "instruments" to populate the list: the constructor and the `concat()`, and this is the third one. Besides, `getActualPaths().add(newPath)` also works out of the box. That is the 4th way)

Reply:
Yes good point, but data empty arrays were added to avoid NPE when calling a method addAll in any programm place.
I’m not sure if your methods can be applied without additional checks, but during initialization when creating and not using seters, most likely yes